### DISCUSSION TITLE
请教一下，路由器上的Openclash+群晖NAS装docker的adguardhome（仅作为NameServer的DNS缓存加速），出现访问过的网站无法访问，未访问的网站可以访问

### DISCUSSION THREAD

**Discussion by Kollata** [vernesong/OpenClash#3965] [General]
我的网络拓扑和配置总结
1. 第一层：光猫（LAN地址：192.168.1.1）
   - 模式：桥接模式
   - DHCP：关闭

2. 第二层：一级路由器（LAN地址：192.168.0.1）
   - WAN口：连接光猫
   - DHCP：开启
   - OpenClash：配置节点订阅，使用Fake-IP（TUN-混合）模式
   - DNS设置：
     - NameServer：指向NAS上的AdGuard Home
     - Fallback：8.8.8.8

3. 第三层：万兆交换机
   - LAN1口：连接一级路由器
   - LAN2口：连接NAS
   - LAN3口：连接电脑

4. 第四层：NAS（IP：192.168.0.106）
   - 安装AdGuard Home（通过Docker）- 作为一级路由器的OpenClash的DNS服务器

问题表现：一旦在openclash的覆写设置下修改 NameServer为adguardhome的DNS配置（地址192.168.0.106；端口1053；UDP）应用后，整个局域网无法访问之前访问过的网址（清理过路由器DNS缓存+电脑缓存，都一样），未访问过的网页可以访问；

手机上WiFi标志显示感叹号，显示网络异常，但是可以上网，就是可以访问未访问过的网页IP
电脑上网络标志正常，一样可以上网，一样可以访问未访问过的网页IP，已访问过的国内网，国外网，例如百度，谷歌全部无法使用

疑点1：是否和分流规则冲突，我注意到我的分流规则最下面是
ruleset=🚀 节点选择,[]DOMAIN-SUFFIX,xn--ngstr-lra8j.com
ruleset=🚀 节点选择,[]DOMAIN-SUFFIX,services.googleapis.cn
ruleset=🚀 节点选择,https://raw.githubusercontent.com/ACL4SSR/ACL4SSR/master/Clash/Ruleset/GoogleCNProxyIP.list
ruleset=🎯 全球直连,[]GEOIP,CN,no-resolve
ruleset=🐟 漏网之鱼,[]FINAL

疑点2：是否和路由器防火墙有关系，又或者说Fake-IP冲突之类的

**Comment by Alinlin918** [comment]
你好，我遇到了和你类似的问题，openwrt是旁路由，在使用openclash时外网不能通过域名访问家里的NAS，关闭openclash就可以恢复访问，找了很多相关视频来看也没找到对的解决方法，不过用password2就没有这个问题，也不知道问题在哪里，真希望有大佬可以帮忙看看，付费也行啊！

**Comment by Alinlin918** [comment]
你好，我遇到了和你类似的问题，openwrt是旁路路由，在使用openclash时外网无法通过域名访问到NAS，关闭openclash就可以恢复访问，找了很多相关视频看，也没找到对的解决方法，不过用password2就没有这个问题，也不知道问题在哪里，真希望有大佬可以帮忙看看，付费也行啊！

**Comment by windy1lee** [comment]
你好，是否解决问题？
